asana/asana.go

```diff
@@ -28,8 +28,18 @@ var defaultOptFields = map[string][]string{
 }
 
 type (
+	//Doer interface used for doing http calls.
+	//Use it as point of setting Auth header or custom status code error handling.
+	Doer interface {
+		Do(req *http.Request) (*http.Response, error)
+	}
```

Comment:
It allows me to use single instance of my configured http.Client and set different Auth headers on each request.

I have configured http.Client and it is property of my own Client struct.
```go
type Client struct {
	httpClient *http.Client
}
```
My service receiving auth token as an argument to request and with Doer interface it is easy to add Auth header with token to request like this:
```go
func (c *Client) withToken(token string) *asana.Client {
	f := func(req *http.Request) (*http.Response, error) {
		req.Header.Set("Authorization", "Bearer "+token)
		return c.httpClient.Do(req)
	}
	return asana.NewClient(asana.DoerFunc(f))
}

// Usage example
func (c *Client) Issue(ctx context.Context, token string, issueID int64) (*Issue, error) {
	t, err := c.withToken(token).GetTask(ctx, issueID, issueFieldsFilter)
	if err != nil {
		return nil, err
	}
	issue := toIssue(t)
	return &issue, nil
}
```